src/models.jl

```diff
@@ -85,25 +86,35 @@ end
 
     kwargs are passed to the regular `Model` constructor.
 """
-ChannelModel(; bcs=ChannelModelBoundaryConditions(), kwargs...) = 
+ChannelModel(; bcs=ChannelModelBCs(), kwargs...) = 
     Model(; bcs=bcs, kwargs...)
           
+#
+# Model initialization utilities
+#
+
 arch(model::Model{A}) where A <: Architecture = A
 float_type(m::Model) = eltype(model.grid)
 add_bcs!(model::Model; kwargs...) = add_bcs(model.boundary_conditions; kwargs...)
 
-function initialize_with_defaults!(eos, tracers, sets...)
+
+function initialize_with_defaults!(eos::EquationOfState, tracers, sets...)
     # Default tracer initial condition is deteremined by eos.
     tracers.S.data.parent .= eos.S₀
     tracers.T.data.parent .= eos.T₀
+    initialize_with_defaults!(sets...)
+    return nothing
+end
 
-    # Set all further fields to 0
+function initialize_with_defaults!(sets...)
```

Comment:
Hmmm, since it sets all field to zero would `initialize_with_zeros!(sets...)` be a better name?

Reply:
Fine with me.